### PR TITLE
script and files needed to update projects.ensembl.org

### PR DESCRIPTION
**Pull request by leannehaggerty**
Here is the script for creating the species.yaml files for the various project pages on projects.ensembl.org.
Also includes current input files (database lists) and output files (species.yaml) and poetry dependencies. 

Happy to discuss in Python meeting.

### CODE REVIEW

**Comment by leannehaggerty** [comment]
Remove instances of username in script!

**Review by williamstark01 — COMMENTED** [review]
This is good work Leanne! And bonus kudos for the nice documentation!

Formatting of the code is the one thing that needs to be fixed. Indentation is always 4 spaces in Python, anything else looks wrong to the eyes of a Pythonista! You can fix this, and improve indentation in general, by simply running the command:
```
black write_yaml.py
```
If you have `pipx` installed, installing `black` takes a single command:
```
pipx install black
```
`pipx` is the recommended way to manage Python CLI applications, you can see how to install it in the Python page:
https://www.ebi.ac.uk/seqdb/confluence/display/ENSGBD/Python#Python-installingPythonCLItools

Other than formatting, I can cherry-pick a few things for improvement, although they are not strictly necessary. But let me know if you want me to do a more thorough code review.

**Comment by leannehaggerty** [comment]
Have run black... I don't like it XD

**Comment by williamstark01** [comment]
Hahaha if it's the indentation it gets some time to get used to it. The splitting of the yaml lines looks ugly to me as well, you can solve it by using f-strings, I'm adding an example in the code.

File: scripts/projects_page/write_yaml.py [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+            + "/geneset/"
+            + date
+            + "/"
+            + species_name
+            + "-"
+            + info_dict["assembly.accession"]
+            + "-"
+            + date
+            + "-genes.gff3.gz\n"
+        )
```
Comment (williamstark01) [inline scripts/projects_page/write_yaml.py:116]: f-string:
```
        yaml += f'  annotation_gtf: {ftp_base}/{species_name}/{info_dict["assembly.accession"]}/geneset/{date}/{species_name}-{info_dict["assembly.accession"]}-{date}-genes.gff3.gz\n'
```

string format method:
```
        yaml += "  annotation_gtf: {}/{}/{}/geneset/{}/{}-{}-{}-genes.gff3.gz\n".format(
            ftp_base,
            species_name,
            info_dict["assembly.accession"],
            date,
            species_name,
            info_dict["assembly.accession"],
            date,
        )
```

I think the second one looks a lot better in this case.

File: scripts/projects_page/write_yaml.py [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+
+        main_species_url = "http://www.ensembl.org/info/about/species.html"
+        main_species_response = requests.get(main_species_url)
+        if info_dict["assembly.accession"] in main_species_response.text:
+            yaml += (
+                "  ensembl_link: https://www.ensembl.org/"
+                + species_name
+                + "/Info/Index\n"
+            )
```
Comment (williamstark01) [inline scripts/projects_page/write_yaml.py:205]: Here an f-strings would look good:
```
            yaml += "  ensembl_link: https://www.ensembl.org/{species_name}/Info/Index\n"
```

**Comment by williamstark01** [comment]
@leannehaggerty I forgot to say, the changes with f-strings or the format method are just suggestions, no need to worry about it if there isn't time to improve the formatting. I could also do a PR with a try to tidy the strings a bit if you like how they look.

**Comment by williamstark01** [comment]
The file `icons.txt` is missing from this directory, was it omitted from the git repo or is it downloaded separately?

Also, I think a `.python-version` file should probably be included in the repo to make recreating the development environment even faster.

**Comment by thibauthourlier** [comment]
This is a comment as a non python user and I don't know the best way to really do it. Having scripts/<project name>/<files> is maybe not the best (I don't really like it is closer to reality). I know this is how it is done in ensembl-analysis but in scripts I would have more expected either <scripts> or <project name>/<scripts>. My preference would be for <project name>/{scripts,yml}/<files> alongside scripts but maybe it is adding to much complexity.
The current option is fine and I am definitely not asking for a change now. Maybe I should create a new issue

**Comment by leannehaggerty** [comment]
> The file `icons.txt` is missing from this directory, was it omitted from the git repo or is it downloaded separately?
> 
> Also, I think a `.python-version` file should probably be included in the repo to make recreating the development environment even faster.

@williamstark01 

txt files added, not sure why they were missing.

what goes in `.python-version`? /nfs/production/panda/ensembl/leanne/.pyenv/versions/3.8.6/envs/project_pages/?

**Comment by williamstark01** [comment]
> what goes in `.python-version`? /nfs/production/panda/ensembl/leanne/.pyenv/versions/3.8.6/envs/project_pages/?

Just `3.8.6/envs/project_pages` is perfect.

**Review by williamstark01 — APPROVED** [review]
Works nicely, happy to merge.

**Comment by williamstark01** [comment]
Sorry @thibauthourlier my bad, I forgot your comment for a second! Do revert the merge if there are changes to be made.

For the comment, I didn't understand what you mean, seems like there are missing words, are you suggesting a change to the directory structure?

**Comment by thibauthourlier** [comment]
@williamstark01 it is definitely missing words but I don't remember what it was. I don't like the structure used for the scripts and the files but not enough to stop the merge or revert it

**Comment by williamstark01** [comment]
@thibauthourlier that's good to hear, for me the directory structure isn't bad or anything but maybe we can improve it, we could discuss when an idea for restructuring comes up, or even as a topic in a future STS.

**Pull request merged**